src/driver/tests/experimental_design_test.py

```diff
@@ -190,6 +228,118 @@ def test_evaluate_with_two_cores(self):
+        parent_dir_path = LocalGSPath("gs://ExperimentalDesignTest/parent")
+        data_design_dir_path = parent_dir_path.joinpath("data_design")
+        experiment_dir_path = parent_dir_path.joinpath("experiments")
+        data_design_dir_path.joinpath("dummy.txt").write_text(
+            "For creating the target directory."
+        )
+        experiment_dir_path.joinpath("dummy.txt").write_text(
+            "For creating the target directory."
+        )
+
+        results = self._evaluate_with_apache_beam(
+            str(data_design_dir_path), str(experiment_dir_path)
+        )
+        # We don't check each column in the resulting dataframe, because these have
+        # been checked by the preceding unit tests.  However, we make a few strategic
+        # probes.
+        self.assertEqual(results.shape[0], 1)
+        self.assertEqual(results["col"][0], 1)
+
+    def _evaluate_with_apache_beam(self, data_design_dir, experiment_dir):
+        num_workers = 2
+        data1 = HeterogeneousImpressionGenerator(
+            1000, gamma_shape=1.0, gamma_scale=3.0
+        )()
+        pdf1 = PublisherData(FixedPriceGenerator(0.1)(data1))
+        data_set = DataSet([pdf1], "dataset")
+        data_design = DataDesign(data_design_dir)
+        data_design.add(data_set)
+
+        msd = ModelingStrategyDescriptor(
+            "single_publisher", {}, "goerg", {}, "pairwise_union", {}
+        )
+        sparams = SystemParameters(
+            [0.5],
+            LiquidLegionsParameters(13, 1e6, 1),
+            np.random.default_rng(),
+        )
+        eparams = ExperimentParameters(
+            PrivacyBudget(1.0, 0.01), 3, 5, "grid", {"grid_size": 5}
+        )
+        trial_descriptors = [TrialDescriptor(msd, sparams, eparams)]
+
+        exp = ExperimentalDesign(
+            experiment_dir,
+            data_design,
+            trial_descriptors,
+            seed=1,
+            cores=num_workers,
+            analysis_type="single_pub",
+        )
+        exp.generate_trials()
+
+        pipeline_args = []
+        pipeline_args.extend(
+            [
+                "--runner=direct",
+                "--direct_running_mode=multi_processing",
+                f"--temp_location={experiment_dir}",
+                f"--direct_num_workers={num_workers}",
+            ]
+        )
+        pipeline_options = PipelineOptions(pipeline_args)
+        return exp.load(use_apache_beam=True, pipeline_options=pipeline_options)
+
+    def test_evaluate_trial_do_fn(self):
+        num_trials = 10
+        seed = 1
+        expected = list(range(num_trials))
+
+        with TestPipeline() as p:
+            output = (
+                p
+                | beam.Create([FakeExperimentalTrial(i) for i in range(num_trials)])
+                | beam.ParDo(EvaluateTrialDoFn(), seed)
+            )
+            assert_that(output, equal_to(expected))
+
+    def test_combine_dataFrame_fn(self):
```

Comment:
Please turn off all of the log messages that this function is generating to stdout.